### PR TITLE
StatisticsCleanup: `stats.kruskal` review

### LINKED ISSUE #639: Statistics Review: kruskal (Trac #112)
_Original ticket http://projects.scipy.org/scipy/ticket/112 on 2006-04-02 by @rkern, assigned to @rkern._

The function `kruskal` in file source:trunk/Lib/stats/stats.py needs review.

Please look over the StatisticsReview guidelines and add your comments below.

### PR DESCRIPTION
**Pull request by aeklant**
closes gh-639

### CODE REVIEW

**Comment by aeklant** [comment]
`nan_policy` addition is missing. I will have to rebase anyway but I will wait to see which of the PRs merges first (this one or the `_chk_asarray` one, where the `nan_policy` implementation is being done) and then rebase and put it in the second one.

Code changes:
* Added empty input handling to return `np.nan, np.nan` if any input is empty.
* Removed unhelpful comments.
* Changed the name of variable `na` to `num_groups` for clarity.

Documentation changes:
* Added reference to the original paper.
* Added `mannwhitneyu`, `f_oneway` and `friedmanchisquare` to `See Also` section.
* Added two examples (taken directly from the regression tests).

ping @rgommers

File: scipy/stats/tests/test_stats.py [hunk 1]
```diff
@@ -3313,6 +3313,14 @@ def test_three_groups(self):
         assert_approx_equal(h, expected)
         assert_approx_equal(p, stats.distributions.chi2.sf(h, 2))
 
+    def test_empty(self):
+        # A test of stats.kruskal with three groups, with ties.
+        x = [1, 1, 1]
+        y = [2, 2, 2]
+        z = []
+        h, p = stats.kruskal(x, y, z)
```
Comment (larsmans) [inline scipy/stats/tests/test_stats.py]: Unused results.

File: scipy/stats/stats.py [hunk 2]
```diff
@@ -4638,33 +4644,55 @@ def kruskal(*args):
 
     References
     ----------
-    .. [1] http://en.wikipedia.org/wiki/Kruskal-Wallis_one-way_analysis_of_variance
+    .. [1] Kruskal, William H. & Wallis, W. Allen, "Use of Ranks in
```
Comment (rgommers) [inline scipy/stats/stats.py]: Can you write the names as ``W.H. Kruskal and W.W. Allen``?

Comment (rgommers) [inline scipy/stats/stats.py]: And ``Volume`` as ``Vol.``? Both minor, but good to keep things consistent with the numpy docstring guidelines.

**Comment by rgommers** [comment]
Looks good to me, can me merged once the minor comments are addressed.

**Comment by aeklant** [comment]
Comments addressed. Nothing major so TravisCI should pass again.

**Comment by rgommers** [comment]
Green & merged. Thanks @aeklant

**Pull request merged**